arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -531,35 +507,34 @@ static void pkvm_vcpu_load(struct pkvm_vcpu *pkvm_vcpu, int cpu)
 		return;
 
 	vcpu = to_kvm_vcpu(pkvm_vcpu);
-
-	/*
-	 * Loading vcpu on a new CPU without putting it from the previous one
-	 * is not supported.
-	 */
-	if (WARN_ON_ONCE(vcpu->cpu != -1 && vcpu->cpu != cpu))
+	loaded_cpu = cmpxchg(&vcpu->cpu, -1, cpu);
+	if (loaded_cpu == -1)
+		/* Not loaded yet */
+		get_pkvm_vm(vcpu->kvm->arch.pkvm.pkvm_vm_handle);
+	else if (WARN_ON_ONCE(loaded_cpu != cpu))
+		/* Already loaded on another CPU */
```

Comment:
nit: I suppose the recommended style is to add {} braces in such cases (i.e. when we have a comment above the statement).

Reply:
checkpath doesn't report coding style issues. And seems the Documentation/process/coding-style.rst doesn't explicitly mention the case for one line comment and another line of code, but this case sounds like to match the "multiple statements" mentioned in the doc?

Adding braces is fine to me. Will update the patch.